### PR TITLE
[WIP] Property pane

### PR DESCRIPTION
**Pull request by Unia**
Cherry-pick of #63 onto master. We still need a new name for `Propertyable`.

### CODE REVIEW

File: src/main/java/nl/tudelft/dnainator/ui/views/View.java [hunk 1]
```diff
@@ -111,4 +116,25 @@ public void zoom(Double zoom, Point2D center) {
...
+	 * @return The last clicked {@link Propertyable}, if any.
+	 */
+	public final Propertyable getLastClicked() {
+		return lastClicked.get();
+	}
+
+	/**
+	 * @return The last clicked property.
+	 */
+	public ObjectProperty<Propertyable> lastClickedProperty() {
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/views/View.java]: Should this be final?

File: src/main/java/nl/tudelft/dnainator/ui/widgets/PropertyPane.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		try {
+			fxmlLoader.load();
+		} catch (IOException e) {
+			new ExceptionDialog(null, e, "Can not load the PropertyPane!");
+		}
+	}
+
+	private void setupHideAnimation(double expandedWidth, int duration) {
+		hide = new Transition() {
+			{ setCycleDuration(Duration.millis(duration)); }
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/PropertyPane.java]: I wonder why check style doesn't whine about this, does it need to be a oneliner?

File: src/main/java/nl/tudelft/dnainator/ui/widgets/PropertyPane.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+			@Override
+			protected void interpolate(double frac) {
+				double curWidth = expandedWidth * (1.0 - frac);
+				setPrefWidth(curWidth);
+				setTranslateX(expandedWidth - curWidth);
+			}
+		};
+		hide.onFinishedProperty().set(new EventHandler<ActionEvent>() {
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/PropertyPane.java]: I'd prefer a lambda expression here.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/PropertyPane.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+ * </p>
+ */
+public class PropertyPane extends VBox {
+	private static final String FXML = "/ui/fxml/propertypane.fxml";
+	private static final int DURATION = 250;
+	private static final int WIDTH = 300;
+	@FXML private Label label;
+	@FXML private VBox vbox;
+	private Animation show;
+	private Animation hide;
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/PropertyPane.java]: Can't the show animation just be reversed to hide the pane?

File: src/main/java/nl/tudelft/dnainator/ui/views/View.java (hunk 1 above)
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/views/View.java]: No it shouldn't

**Comment by pvanagtmaal** [comment]
#102 requires this PR to be redesigned.

**Comment by pvanagtmaal** [comment]
A more recent and improved PR #114 replaces this.

**Pull request closed without merging**